src/orquestra/opt/optimizers/scipy_optimizer.py

```diff
@@ -14,6 +14,77 @@
 )
 from ..history.recorder import RecorderFactory
 from ..history.recorder import recorder as _recorder
+from .pso.continuous_pso_optimizer import _get_bounds_like_array
+
+
+class _CostFunctionWithBestValueType(type):
+    def __instancecheck__(cls, obj: object) -> bool:
+        if (
+            isinstance(obj, Callable)
+            and hasattr(obj, "best_value")
+            and hasattr(obj, "best_params")
+        ):
+            return True
+        return False
```

Comment:
This is a nice little piece of code! But I'm afraid it might be needlessly opaque since isinstance is only used once here and just for an assert statement. metaclasses are considered "heavy machinery" so it's best not to use them when they could be replaced with a simple if statement.

Reply:
If I remove this to check the new instance check, I get some linting problems 😢

Reply:
Yes, but since you only use this isinstance once, it's not really worth changing the __instancecheck__ method right?

Reply:
I had to do this trick for the tensor train optimiser PR too #31. Do you want me to instead have a custom `isinstanceCostFunctionWithBestValue`, or something like that?

Reply:
No, it's probably fine. Not worth holding up the PR for this.